README.md

```diff
@@ -16,3 +16,142 @@ bower install purescript-argonaut-core
 ## Documentation
 
 Module documentation is [published on Pursuit](http://pursuit.purescript.org/packages/purescript-argonaut-core).
+
+## Tutorial
+
+Some of Argonaut's functions might seem a bit arcane at first, so it can help
+to understand the underlying design decisions which make it the way it is.
+
+One approach for modelling JSON values would be to define an ADT, like this:
+
+```purescript
+data Json
+  = JNull
+  | JBoolean Boolean
+  | JArray (Array Json)
+  | JObject (StrMap Json)
+  [...]
+```
+
+And indeed, some might even say this is the obvious approach.
+
+Because Argonaut is written with the compilation target of JavaScript in mind,
+it takes a slightly different approach, which is to reuse the existing data
+types which JavaScript already provides. This way, the result of JavaScript's
+`JSON.stringify` function is already a `Json` value, and no extra processing is
+needed before you can start operating on it. This ought to help your program
+both in terms of speed and memory churn.
+
+Much of the design of Argonaut follows naturally from this design decision.
+
+### Introducing Json values
+
+(Or, where do `Json` values come from?)
+
+Usually, a `Json` value will be introduced into your program via either the FFI
+or via the construction functions in `Data.Argonaut.Core`. Here are some
+examples:
+
+```javascript
+// In an FFI module.
+exports.someNumber = 23.6;
+exports.someBoolean = false;
+exports.someObject = {people: [{name: "john"}, {name: "jane"}], 
+```
+
+```purescript
+foreign import someNumber :: Json
+foreign import someBoolean :: Json
+foreign import someObject :: Json
+```
+
+Generally, if a JavaScript value could be returned from a call to
+`JSON.stringify`, it's fine to import it from the FFI as `Json`. So, for
```

Comment:
I think what you mean is something like:

any data structure that can be an actual parameter to `JSON.stringify` and that then is converted into a valid JSON string can also immediately imported from the FFI as type `Json`